lib/addons/helm-addon/index.ts

```diff
@@ -23,4 +25,20 @@ export abstract class HelmAddOn implements spi.ClusterAddOn {
         const chart = {...this.props, ...{ values, wait, timeout, createNamespace} };
         return kubectlProvider.addHelmChart(chart);
     }
-}
\ No newline at end of file
+}
+//below is my addon approach method for when I get too it. Leaving for later
+function validateInputAddonConstraints(props: HelmAddOnProps){
+    try{
+        const test = z.string().max(addonConstraints.addonNameMaxLength);
+        console.log(props.name + "-=-=-=-=-=");
+        test.parse(props.name);
+    }
+    catch(e){
+        throw new Error('Addon Names must be no more than 128 characters long!');
+    }
+}
+
+const addonConstraints = {
```

Comment:
I assume scratchpad code? let's drop it and replace with the data structure that defines constraints with our constraints framework.

Reply:
Also from before we changed to the generic way and I haven't gone back to this file yet. I'll be sure to fix up this file and remove the old stuff.